### ISSUE TITLE
Issue with opening psbt on coldcard for 3-of-2 wallet transacation

### ISSUE THREAD

**Issue by pleblira** [cryptoadvance/specter-desktop#1539]
Hi everyone. 

I’m trying to create a multisig with a coldcard, a ledger, and an electrum wallet, and running into problems when exporting the psbt file to the cold card. When I open it on the cold card I get the following error.

`Failure - invalid PSBT - deriv 0 != 4 xpub depth (xfp=mywalletfingerprint)`

The electrum multisig wallet was created using a BIP39 seed (not an Electrum seed) and the zpub imported successfully into Specter. I was able to import the wallet .txt file back into the Coldcard successfully and was able to sign the transaction on both Ledger and the Electrum. Just wasn't able to sign the PSBT with the Coldcard.

If I open the transaction that was meant for the Coldcard on the Electrum wallet (not that it would do anything), I get a similar error: “Electrum was unable to parse your transaction: SerializationError('PSBT global xpub has mismatching depth (4) and derivation prefix len (0)’)”

Not sure I'm doing something wrong here, as both Ledger and Electrum are able to sign, the wallet file was imported successfully, and the error message seems to be consistent on both Coldcard and Electrum.

One last note: If I create a 2-of-2 wallet with just the Ledger and the Coldcard, it signs fine. Happy to add more information here in case it's helpful. 

Thanks for any help in advance!

**Comment by pleblira** [comment]
Could it be an issue that when I look at the PDF backup file, it doesn't include the derivation paths?

![2022-01-28 at 9 51 AM](https://user-images.githubusercontent.com/91032628/151568150-1b796b37-0352-4a1e-bb67-0b76ce72e43f.png)

**Comment by k9ert** [comment]
Yes, definitely! This is an issue with the derivation-path. We had it here and i tracked down "deriv 0" to zero slashes in the derivationpath.
https://github.com/cryptoadvance/specter-desktop/issues/1452#issuecomment-1006685168

@relativisticelectron any idea what role electrum could play here?

**Comment by relativisticelectron** [comment]
Hi, 
1. Question: Did you create the _unsigned_ psbt with electrum or specter?
2. Can you double check if the derivation path `wallet.get_keystores()[0].get_derivation_prefix()` and fingerprint `wallet.get_keystores()[0].get_root_fingerprint()`  (with `[0]`, `[1]`, and `[2]`)   is saved in electrum?  If it is not there you can add both using: https://github.com/spesmilo/electrum/issues/5955#issuecomment-724321441
3. Why is the derivation path missing in specter? Did you import the "electrum signer" from the electrum wallet, or create it manually?  After ensuring that derivation path and fingerprint are in electrum (point 2.), you can try reimporting the "electrum signer" in specter.

**Comment by pleblira** [comment]
Hi, thanks for looking into it!

1. The unsigned psbt was created with Specter
2. If I recreate the multisig on Electrum by using the BIP39 seed + the Zpubs from the Ledger and Coldcard, getting derivation paths and fingerprints will only return anything for the wallet that I used the seeds. Ledger and Coldcard will return empty values (example below). Since the transaction was created on Specter, though, I'm not sure this relates to this issue, does it?
3. I imported the Electrum wallet into Specter by pasting the Zpub into Specter. I couldn't find a way to edit the derivation paths for the wallet either before or after adding it to Specter... unless I'm missing something. Below is an example.


![2022-02-08 11 00 51](https://user-images.githubusercontent.com/91032628/153027050-fe8e49ae-8089-40a3-8ffc-66bd1eadb032.gif)
![2022-02-08 at 10 49 AM](https://user-images.githubusercontent.com/91032628/153023782-7cee0c28-9929-4f67-819c-85874bd861ab.png)

**Comment by relativisticelectron** [comment]
Try 

> If it is not there you can add both using: [Add full BIP32 derivation to PSBT spesmilo/electrum#5955 (comment)](https://github.com/spesmilo/electrum/issues/5955#issuecomment-724321441)  You can see the electrum console in the menu  View --> Show console.

and reimporting the electrum wallet into specter.

**Comment by relativisticelectron** [comment]
Ok, after studying the code  https://github.com/cryptoadvance/specter-desktop/blob/10065f10a1674cf81baa80696d287f260b669b91/src/cryptoadvance/specter/templates/device/new_device/new_device_keys.jinja#L182   there is a way to add fingerprint, derivation and xpub. You have to format all as `[fingerprint/48h/1h/0h/2h]xpub`  and paste it into the "paste xpub" field.  But this is not user friendly at all.

**Comment by relativisticelectron** [comment]
Lesson learned:  The "Add device" --> "Electrum" is impossible to use for almost anybody.  This has to be redesigned.

**Comment by k9ert** [comment]
Thank you so far! I guess we have at least a reasonable workaround and maybe we should deactivate the electrum-device for now? I'm not sure what value it even brings in here other than being buggy.

**Comment by k9ert** [comment]
I think we can close this as we solved it with #1622 .
